go/exocom/src/client_registry/client_registry_test.go

```diff
@@ -0,0 +1,106 @@
+package clientRegistry_test
+
+import (
+	"github.com/Originate/exocom/go/exocom/src/client_registry"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ClientRegistry", func() {
+	var registry *clientRegistry.ClientRegistry
+
+	Describe("can send", func() {
+		BeforeEach(func() {
+			serviceRoutes := `{"role 1": {"sends": ["message 3 name"]}}`
+			var err error
+			registry, err = clientRegistry.NewClientRegistry(serviceRoutes)
+			Expect(err).To(BeNil())
+		})
+		It("returns true if the routing table includes the message name", func() {
+			result := registry.CanSend("role 1", "message 3 name")
+			Expect(result).To(BeTrue())
+		})
+		It("returns false if the routing table does not include the message name", func() {
+			result := registry.CanSend("role 1", "message 2 name")
+			Expect(result).To(BeFalse())
+		})
+
+	})
+	Describe("subscribers for", func() {
+		BeforeEach(func() {
+			serviceRoutes :=
+				`{
+					"role 1": {
+						"receives": ["message 3 name"],
+						"internalNamespace": "namespace"
+					}
+				}`
+			var err error
+			registry, err = clientRegistry.NewClientRegistry(serviceRoutes)
```

Comment:
Why are we providing the routes as a string here? That seems strange. I would expect we parse config files into Go structs.